### DISCUSSION TITLE
Is it possible to open and view the _dbs.couch under the data directory

### DISCUSSION THREAD

**Discussion by zssai** [apache/couchdb#3942] [Q&A]
After installation, under the data directory, there are four databases, _dbs.couch, _nodes.couch, _replicator.couch and _users.couch. Is it possible to open and view the _dbs.couch database

**Comment by nickva** [comment]
@zssai yes, you may view those databases. They are node-local (not clustered), so you could access them via `/_node/_local/_dbs` for example. Instead of `_local` you may use a specific node name `couchdb01@myhost.net` too.

`_dbs` would contain shards for clustered databases. It's a database of database docs :-)

```
http put $DB/mydb
```

```
http $DB/_node/_local/_dbs/mydb
HTTP/1.1 200 OK

{
    "_id": "mydb",
    "_rev": "1-275115469f6f9a8469008e5e5a929d2c",
    "by_node": {
        "node1@127.0.0.1": [
            "00000000-7fffffff",
            "80000000-ffffffff"
        ]
    },
    "by_range": {
        "00000000-7fffffff": [
            "node1@127.0.0.1"
        ],
        "80000000-ffffffff": [
            "node1@127.0.0.1"
        ]
    },
    "changelog": [
        [
            "add",
            "00000000-7fffffff",
            "node1@127.0.0.1"
        ],
        [
            "add",
            "80000000-ffffffff",
            "node1@127.0.0.1"
        ]
    ],
    "props": {},
    "shard_suffix": [
        46,
        49,
        54,
        52,
        53,
        55,
        50,
        49,
        48,
        48,
        56
    ]
}
```

A word of warning: modifying them by hand unless you're aware of the CouchDB internals may render your cluster unusable and you may lose data.

**Comment by zssai** [comment]
Hello, Nick
Thank you for your help. I will test it and reply to you later.

Best Regards
 
From: Nick Vatamaniuc
Date: 2022-02-25 00:45
To: apache/couchdb
CC: zssai; Mention
Subject: Re: [apache/couchdb] Is it possible to open and view the _dbs.couch under the data directory (Discussion #3942)
@zssai yes, you may view those databases. They are node-local (not clusters) so you could access them via /_node/_local/_dbs for example.
_dbs would contain shards for clustered databases. It's a database of database docs :-)
http put $DB/mydb
http $DB/_node/_local/_dbs/mydb
HTTP/1.1 200 OK
{
    "_id": "mydb",
    "_rev": "1-275115469f6f9a8469008e5e5a929d2c",
    "by_node": {
        ***@***.***": [
            "00000000-7fffffff",
            "80000000-ffffffff"
        ]
    },
    "by_range": {
        "00000000-7fffffff": [
            ***@***.***"
        ],
        "80000000-ffffffff": [
            ***@***.***"
        ]
    },
    "changelog": [
        [
            "add",
            "00000000-7fffffff",
            ***@***.***"
        ],
        [
            "add",
            "80000000-ffffffff",
            ***@***.***"
        ]
    ],
    "props": {},
    "shard_suffix": [
        46,
        49,
        54,
        52,
        53,
        55,
        50,
        49,
        48,
        48,
        56
    ]
}

A word of warning: modifying them by hand unless you're aware of the CouchDB internals may render your cluster unusable and you may lose data.
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
